service-registry/src/main/java/io/servicecomb/serviceregistry/cache/InstanceCacheManagerOld.java

```diff
@@ -149,6 +150,11 @@ public void onInstanceUpdate(MicroserviceInstanceChangedEvent changedEvent) {
       switch (changedEvent.getAction()) {
         case CREATE:
         case UPDATE:
+          if (changedEvent.getInstance().getStatus() != MicroserviceInstanceStatus.UP) {
+            instMap.remove(changedEvent.getInstance().getInstanceId());
+            cacheMap.put(key, new InstanceCache(appId, microserviceName, version, instMap));
+            break;
+          }
           instMap.put(changedEvent.getInstance().getInstanceId(), changedEvent.getInstance());
```

Comment:
how about putting this line in the else clause to avoid duplicate statement of line 155 and 159?

Reply:
done